packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -821,6 +822,81 @@ void main() {
       binding.postTest();
     });
   });
+
+  group('Accessibility announcements testing API', () {
+    testWidgets('Returns last announcement', (WidgetTester tester) async {
+      await SemanticsService.announce('announcement 1', TextDirection.ltr);
+      await SemanticsService.announce('announcement 2', TextDirection.rtl,
+          assertiveness: Assertiveness.assertive);
+
+      final CapturedAccessibilityAnnouncement last = tester.getLastAnnouncement()!;
+      expect('announcement 2', equals(last.message));
+      expect(TextDirection.rtl, equals(last.textDirection));
+      expect(Assertiveness.assertive, equals(last.assertiveness));
+    });
+
+    testWidgets('Returns the list of announcements', (WidgetTester tester) async {
+      await SemanticsService.announce('announcement 1', TextDirection.ltr);
+      await SemanticsService.announce('announcement 2', TextDirection.rtl,
+          assertiveness: Assertiveness.assertive);
+      await SemanticsService.announce('announcement 3', TextDirection.rtl);
+
+      final List<CapturedAccessibilityAnnouncement> list = tester.takeAnnouncements()!;
+      expect(3, equals(list.length));
```

Comment:
A more idiomatic way to test length is: `expect(list, hasLength(3));` (it produces better error messages when the test fails)